### ISSUE TITLE
Manual partitioning: wrong initial format checkbox state

### ISSUE THREAD

**Issue by jpnurmi** [canonical/ubuntu-desktop-installer#1828]
Originally reported by @ogayot:

> When I enter the manual partitioning screen, my existing partition(s) has/has the format checkbox checked. As soon as I click on the partition on or the freespace, the format checkbox gets unchecked.  Is this a known issue? It does not seem critical but feels a bit scary 

| Before select | After select |
|---|---|
| ![format](https://user-images.githubusercontent.com/140617/232803871-f7bc3877-ddbe-42ee-a532-3cad829a7e05.png) | ![free](https://user-images.githubusercontent.com/140617/232803816-24a99d26-af38-4529-a76c-3e96a728b3c1.png) |

Response by @jpnurmi:

> looks like storage & original storage config requests sent in parallel, only one notifies the GUI about changes so the GUI might not update correctly depending on the completion order. my fault, luckily just wrongly presented GUI state

**Comment by ogayot** [comment]
I confirm this is working with latest daily build, thanks!

On the other hand, a similar issue seems to occur when new partitions are discovered after clicking on "revert". I think it's more a edge case and clearly lower priority. But here are the steps to reproduce:

* Navigate to the manual partitioning screen
* Insert a USB stick (that is already partitioned)
* Wait a few seconds
* Click on "Revert"
* The new partitions (from the USB stick) will appear and will have the "Format" checkbox checked.

@jpnurmi is it better if I open a separate issue?
